### ISSUE TITLE
Can't get i18n-js to work with "module: esnext"

### ISSUE THREAD

**Issue by mikew** [DefinitelyTyped/DefinitelyTyped#22964]
- [x] I tried using the `@types/xxxx` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @ypresto 

```typescript
import i18nJs = require('i18n-js')
```

Yields

```
[ts] Import assignment cannot be used when targeting ECMAScript modules. Consider using 'import * as ns from "mod"', 'import {a} from "mod"', 'import d from "mod"', or another module format instead
```

While rewriting it brings up a different error:

```typescript
import * as i18nJs from 'i18n-js'
i18nJs.defaultLocale = 'en'
```

Yields

```
Cannot assign to 'defaultLocale' because it is a constant or a read-only property.
```

Even if I ignore the initial "Import assignment cannot be used when targeting ECMAScript modules", since it doesn't seem to stop builds, there's yet another error:

```typescript
import i18nJs = require('i18n-js')
console.log(i18nJs)
```

Yields

```
i18nJs is not defined
```

**Comment by ypresto** [comment]
@mikew

> Import assignment cannot be used when targeting ECMAScript modules

`i18n-js` is not ES2015 module, so you can't use `es2015` / `es6` / `esnext` for  `module` compiler option in `tsconfig.json`. It is default when you set `target` to other than `es3` or `es5`.

> --module-m | string | target === "ES3" or "ES5" ? "CommonJS" : "ES6"

https://www.typescriptlang.org/docs/handbook/compiler-options.html

**Comment by ypresto** [comment]
TL;DR: You can use `Import Assignment` when you set `module` to `commonjs`.

**Comment by orta** [comment]
Hi thread, we're moving DefinitelyTyped to use [GitHub Discussions](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/53377) for conversations the `@types` modules in DefinitelyTyped.

To help with the transition, we're closing all issues which haven't had activity in the last 6 months, which includes this issue. If you think closing this issue is a mistake, please pop into the [TypeScript Community Discord](https://discord.gg/typescript) and mention the issue in the `definitely-typed` channel.

**Comment by orta** [comment]
Hi thread, we're moving DefinitelyTyped to use [GitHub Discussions](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/53377) for conversations the `@types` modules in DefinitelyTyped.

To help with the transition, we're closing all issues which haven't had activity in the last 6 months, which includes this issue. If you think closing this issue is a mistake, please pop into the [TypeScript Community Discord](https://discord.gg/typescript) and mention the issue in the `definitely-typed` channel.
